neurodamus/metype.py

```diff
@@ -156,7 +156,15 @@ def _instantiate_cell(self, gid, etype_path, emodel, morpho_path, meinfos_v6, de
         add_params = meinfos_v6.add_params or (keep_axon,)  # Keep axon incompatible with add_params
 
         logging.debug("Loading Gid %d: emodel: %s, Morphology: %s", gid, emodel, morpho_file)
-        self._cellref = EModel(gid, morpho_path, morpho_file, *add_params)
+        try:
+            # For this step, do not call mpi_abort in neuron and let neurodamus handle and abort,
+            # NB: Do not re-raise as ConfigurationError, neurodamus doesn't call mpi_abort so hangs
+            old_flag = Nd.pc.mpiabort_on_error(0)
```

Comment:
Shouldnt this be the default behaviour of neuron when called from Neurodamus?

Reply:
Yes, I would like to start a discussion on it. 

I also think we should let neurodamus handle the exceptions from neuron. The only concern is that if letting neurodamus calls the mpi_abort, we will not have the stack traceback from neuron for debugging, is that acceptable?

Reply:
My understand is that @WeinaJi added these lines after observing that it wasn't possible to catch the exception, because something had already called `MPI_Abort`.

I agree that this should be the default. I don't think it should be part of this PR, we can create an issue to "Review MPI_Abort during exception handling" or similar.

Weina, did you check that the above lines are correct, e.g. by inserting an exception that only happens on one MPI rank and observing that all MPI ranks are correctly killed? If yes, this, to me, looks good to merge.

Reply:
I agree to decide whether using`Nd.pc.mpiabort_on_error(0)` as default in another PR, probably including more developers in the discussion. Let's focus on the emodel loading in this PR.

@1uc , yes, I've been testing with multiple ranks with 1 error cell. The exception is raised in one rank and neurodamus kills all the processes using the function `show_exception_abort`.

Reply:
Very good. Then I'm out of idea what could go wrong.